app/src/main/java/com/openclassrooms/realestatemanager/repository/PictureRepository.kt

```diff
@@ -0,0 +1,50 @@
+package com.openclassrooms.realestatemanager.repository
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.map
+import com.openclassrooms.realestatemanager.api.PictureDatabaseApi
+import com.openclassrooms.realestatemanager.api.PictureStorageApi
+import com.openclassrooms.realestatemanager.model.Picture
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class PictureRepository (
+    private val pictureDatabaseApi: PictureDatabaseApi,
+    private val pictureStorageApi: PictureStorageApi
+    ) {
+
+    private val pictureListLiveData = pictureDatabaseApi.getPictureListFlow().asLiveData()
+
+    fun getPictureListLiveData() = pictureListLiveData
+
+    fun getPictureListByEstateIdLiveData(estateId: Long): LiveData<List<Picture>> =
+        pictureListLiveData.map {
+                initialPictureList -> initialPictureList.let {
+                    notNullPictureList -> notNullPictureList.filter {
+                        picture -> picture.estateId == estateId }
+        }
+    }
+
+    fun addPictureList(pictureList: List<Picture>) {
+        CoroutineScope(Dispatchers.IO).launch {
+            for (picture in pictureList) {
+                addPicture(Picture(picture.id, picture.estateId, picture.getUri(), picture.description))
```

Comment:
Pourquoi reconstruis tu un objet Picture ? Peux tu l'utiliser directement ?

```suggestion
                addPicture(picture)
```

Reply:
Oui c'est vrai :)

Reply:
https://github.com/Tgarfr/Project-9-RealEstateManager/pull/8/commits/43c5203ca055317372afc562ac3c658b40f97113